meta-mender-demo/recipes-mender/mender-modules-gen/mender-modules-gen.bb

```diff
@@ -0,0 +1,74 @@
+DESCRIPTION = "Mender demo Update Modules Artifact generators"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
+
+RDEPENDS_${PN} = "bash"
+
+S = "${WORKDIR}/git"
+
+inherit allarch
+
+BBCLASSEXTEND = "native"
+
+def mender_autorev_if_git_version(d):
+    version = d.getVar("PREFERRED_VERSION_pn-mender-client")
+    if not d.getVar("EXTERNALSRC") and version is not None and "git" in version:
+        return d.getVar("AUTOREV")
+    else:
+        return "f6ffa190892202263fdb75975059fbb201adab6a"
+
+SRCREV = '${@mender_autorev_if_git_version(d)}'
+
+# TODO: take into account build tags and other "-git" expressions
+def mender_modules_gen_version_from_mender_version(d, srcpv):
+    import re
+    pref_version = d.getVar("PREFERRED_VERSION_pn-mender-client")
```

Comment:
@lluiscampos: Ref what we talked about earlier: Here it appears to use `pn-`, but I thought without `-pn` was the correct approach. In the end I think this might stem from having initially misunderstood it, but `pn-` also happened to work, so this is what we have used in our builds. The whole issue might be a "poteito, potato" discussion at this point.